File: Problems/Dynamic-programming/Climbing-Stairs/Climbing-Stairs.cpp
#include <iostream>

// Using bottom-up approach - Time: O(n)

class Solution {
public:
    int climbStairs(int n)
    {
        int ans = 1, prev = 1;
        while (--n)
        {
            ans += prev;
            prev = ans - prev;
        }

        return ans;
    }
};

int main()
{
    int n = 5;
    std::cout << "Input: " << n << std::endl;

    Solution sol;
    int result = sol.climbStairs(n);

    std::cout << "Output: " << result << std::endl;

    return 0;
}

File: Problems/Dynamic-programming/Climbing-Stairs/Climbing-Stairs.cs
using System;

// Using bottom-up approach - Time: O(n)

public class Solution
{
    public int ClimbStairs(int n)
    {
        int ans = 1;
        int prev = 1;

        while (--n > 0)
        {
            int temp = ans;
            ans += prev;
            prev = temp;
        }

        return ans;
    }
}

class Program
{
    static void Main(string[] args)
    {
        int n = 5;

        Console.WriteLine("Input: " + n);

        Solution sol = new Solution();
        int result = sol.ClimbStairs(n);

        Console.WriteLine("Output: " + result);
    }
}